connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -582,4 +625,170 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+
+            offsetSeqNum = metricGroup.metrics().sensor("offset-seq-number");
+            offsetSeqNum.add(metricGroup.metricName("offset-commit-seq-no",
+                    "The current sequence number for offset commits"),
+                    new Value());
+
+            offsetCompletion = metricGroup.metrics().sensor("offset-commit-completion");
+            offsetCompletion.add(metricGroup.metricName("offset-commit-completion-rate",
+                    "The average per-second number of offset commit completions that were completed successfully."),
+                    new Rate());
+            offsetCompletion.add(metricGroup.metricName("offset-commit-completion-total",
+                    "The total number of offset commit completions that were completed successfully."),
+                    new Total());
+
+            offsetCompletionSkip = metricGroup.metrics().sensor("offset-commit-completion-skip");
+            offsetCompletionSkip.add(metricGroup.metricName("offset-commit-completion-skip-rate",
+                    "The average per-second number of offset commit completions that were received too late and skipped/ignored."),
+                    new Rate());
+            offsetCompletionSkip.add(metricGroup.metricName("offset-commit-completion-skip-total",
+                    "The total number of offset commit completions that were received too late and skipped/ignored."),
+                    new Total());
+
+            putBatchTime = metricGroup.metrics().sensor("put-batch-time");
+            putBatchTime.add(metricGroup.metricName("put-batch-max-time-ms",
+                    "The maximum time taken by this task to put a batch of sinks records."),
+                    new Max());
+            putBatchTime.add(metricGroup.metricName("put-batch-avg-time-ms",
+                    "The average time taken by this task to put a batch of sinks records."),
+                    new Avg());
+        }
+
+        void computeSinkRecordLag() {
+            Map<TopicPartition, OffsetAndMetadata> consumed = this.consumedOffsets;
+            Map<TopicPartition, OffsetAndMetadata> committed = this.committedOffsets;
+            activeRecords = 0L;
+            for (Map.Entry<TopicPartition, OffsetAndMetadata> committedOffsetEntry : committed.entrySet()) {
+                final TopicPartition partition = committedOffsetEntry.getKey();
+                final OffsetAndMetadata consumedOffsetMeta = consumed.get(partition);
+                if (consumedOffsetMeta != null) {
+                    final OffsetAndMetadata committedOffsetMeta = committedOffsetEntry.getValue();
+                    long consumedOffset = consumedOffsetMeta.offset();
+                    long committedOffset = committedOffsetMeta.offset();
+                    long diff = consumedOffset - committedOffset;
+                    activeRecords += diff;
+                }
+            }
+            activeRecords = Math.max(activeRecords, 0L);
```

Comment:
I'm not sure it's valuable per-partition. How can you get negative numbers? Can offsets be negative? If so, wouldn't it suffice to add the absolute value of `diff`?

Reply:
sorry, that was unclear. i meant put the `Math.max` inside the loop and check `diff` instead of the aggregated `activeRecords`. you shouldn't get negative numbers, but if a connector does something wonky with the offset commits, it could happen that committed offsets are ahead of consumed offsets.

Reply:
Ack